contracts/HubPool.sol

```diff
@@ -607,6 +609,7 @@ contract HubPool is HubPoolInterface, Testable, Lockable, MultiCaller, Ownable {
         int256[] memory netSendAmounts,
         int256[] memory runningBalances,
         uint8 leafId,
+        bool relayToSpokePool,
```

Comment:
What was the reasoning for using a bool over an index? Would something like `perChainLeafIndex` be more clear to someone creating a bundle?

Reply:
I just don't see any utility for the index. If you could give me an example of how an index would be useful to the off-chain data worker, then I'd change to an index. But I can't think of any examples where the index is useful.

Reply:
I think it's just easier to understand since an index obviously increments for each leaf for a chain. Whereas you have to know that this bool goes true on the first element and then false on every following element. I think the bool makes a lot of sense _in the context of the HubPool_, but to a relayer, this bool might have less meaning since they aren't working from within the HubPool's send logic. Maybe to reverse the question, what's the reasoning for using a bool over an index?

Not strongly opinionated on this, however. I'm cool with whatever you think!

Reply:
Hm ok i'll change to an index since it doesn't seem to matter too much and it gives us a bit more flexibility

Reply:
Ok changed @mrice32 but I don't love the name `groupIndex` nor the `perChainleafIndex`. Any other ideas?

Reply:
Yeah, the naming is difficult. I think groupIndex is totally fine, though, especially since you documented it so well.